bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -356,4 +364,14 @@ public Set<AddonInfo> getSuggestedAddons() {
     public String getServiceName() {
         return SERVICE_NAME;
     }
+
+    private boolean isAddonInstalled(String addonId) {
```

Comment:
Maybe we should move this to BaseAddonFinder. It would allow to use this in other add-on finders as well.

Reply:
Sure, if it is needed, I can do so. I had the impression, that only IP finder will skip searching if an addon is already installed.

Should we move it, although it introduces some overhead to the other finders? OSGI would then call register all addonservices to BaseAddonFinder using
`@Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
    protected void addAddonService(AddonService featureService) {`...

If we want to share it, maybe extending AddonResource would be an option.

@wborn Maybe you could comment on this as well....

Reply:
> If we want to share it, maybe extending AddonResource would be an option.

The `AddonResource` is in the org.openhab.core.io.rest.core bundle which is not the type of bundle you want other bundles to depend on. That's because that bundle depends on JAX-RS, Swagger etc.

It might be better to add something reusable to the org.openhab.core.addon bundle. It already provides the `AddonService` interface so anything that needs to interact with `AddonService`s already depends on it. E.g. an `AddonServiceRegistry` could be added there that makes the logic of the  `AddonResource` more reusable. The `AddonResource` could then also use it.

Reply:
Ok, good to know. Actually, an ```AddonServiceRegistry``` was the first thing I was looking for - and wondered that it is not yet there.

However, hhis is clearly out of scope of this PR.
Can we move forward with this one as it is for now?